dart/lib/src/sentry_options.dart

```diff
@@ -327,6 +328,19 @@ class SentryOptions {
   /// The default is 3 seconds.
   Duration? idleTimeout = Duration(seconds: 3);
 
+  final _causeExtractorsByType = <Type, ExceptionCauseExtractor>{};
+
+  Map<Type, ExceptionCauseExtractor> get causeExtractorsByType =>
+      _causeExtractorsByType;
+
+  void addTypedExceptionCauseExtractor(
+      Type type, ExceptionCauseExtractor extractor) {
+    _causeExtractorsByType[type] = extractor;
+  }
```

Comment:
I believe if you change the `ExceptionCauseExtractor` to 
```dart
class ExceptionCauseExtractor<T> {
  ExceptionCause? cause(T error) {
    return null;
  }

  Type exceptionType => T;
}
```
you can change this method to 
```suggestion
  void addTypedExceptionCauseExtractor(ExceptionCauseExtractor extractor) {
    _causeExtractorsByType[extractor.exceptionType] = extractor;
  }
```

Reply:
I wonder if this method should disallow `dynamic` and `Object`?

Reply:
It'd work, indeed:

`Type get exceptionType => T;`

Just missing the `get`

Reply:
Great idea!